src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader_search.cs

```diff
@@ -9,16 +9,16 @@ public static partial class BufferReaderExtensions
     {
         public static bool TryReadUntill(ref BufferReader reader, out ReadOnlySequence<byte> bytes, byte delimiter)
         {
-            var copy = reader;
-            var start = reader.Position;
+            BufferReader copy = reader;
```

Comment:
FYI @JeremyKuhne, this change improves the perf by almost 2x.

We are still roughly 5x slower than the single span case, but a lot of the performance cost is from this method:
- struct copy
- ReadOnlySequence slicing
- Two calls to GetPosition

|                                            Method |   TestCase |        Mean |      Error |     StdDev |
|-------------------------------------------------- |----------- |------------:|-----------:|-----------:|
| **ReaderSystemTextJsonLabMultiSpanSequenceEmptyLoop** |  **BasicJson** |  **2,661.8 ns** |  **23.501 ns** |  **21.983 ns** |
| **ReaderSystemTextJsonLabMultiSpanSequenceEmptyLoop** | **HelloWorld** |    **525.7 ns** |   **4.750 ns** |   **4.211 ns** |
| **ReaderSystemTextJsonLabMultiSpanSequenceEmptyLoop** |   **Json400B** |  **3,892.2 ns** |  **74.134 ns** |  **69.345 ns** |
| **ReaderSystemTextJsonLabMultiSpanSequenceEmptyLoop** |    **Json4KB** | **31,168.4 ns** | **297.191 ns** | **277.992 ns** |